source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+   Mnemonics are case-sensitive.
+
+#. Menu items referencing
+   commands that can not be found
+   are disabled.
+
+#. Menu items can be hidden or disabled
+   by the command.
+
+.. TODO add refs to Command methods
+
+
+.. _menu-separators:
+
+Separators
+**********
+
+Separators are menu items
+with the caption ``-``
+and no sub-menu.
+They are commonly used
+to group menu items with similar purposes
+or with a shared thematic.
+Separators can not call commands,
+but the presence of a sub-menu
+converts the separator item into a regular item
+with a single hyphen as its caption.
+
+For rendering,
+multiple consecutive separators are reduced to one
+and separators at the beginning or the end of a menu
+are not displayed.
+
+
+.. _menu-merging:
+
+Menu Merging
+============
+
+Generally,
+``.sublime-menu`` files are loaded in package load order.
+Menu files with the same name are concatenated
+by extending the menu item list
+with all items in the file,
+unless IDs are involved.
+See below for this case.
+
+Menu files in the same package are loaded lexicographically
+in the root folder,
+after which the sub folders are traversed similarly.
+
+As a special case,
+menu items from the *User* package
+in the standard non-ID section
+are always inserted after any standard items
+from other packages.
+
+
+.. _item-ids:
+
+Item IDs
+========
+
+When a menu item specifies an ID,
+a separate section within the menu is searched for
+and, if it does not exist,
+created at the very end of the menu.
+*This is a forward lookup only.*
+The ID determines the section's name
+and the menu item with the ID
+will be the first item in this section.
+All following items in the file
+will then be appended to the ID's section,
+until another item with an ID is specified.
+
+If two menu items
+from different ``.sublime-menu`` files
+reference the same item via ID,
+Sublime Text will override the previously defined item's parameters
+with the new parameters,
+if there are any.
+Child elements in a sub menu are appended
+and do not override.
+All following items are then appended to the ID's section,
+until another item with an ID is specified.
+
+Common practice is
+to assign IDs to separators
+and items having a sub-menu,
+to allow other packages or the user themselves
+to easily customize the menu.
+Separators always mark the beginning of a section,
+while it is required to address the exact same menu item
+in order to append items to its sub-menu.
+
+.. note::
+
+   Due to the strict forward lookup
+   it is possible to have
+   *multiple different items with the same ID*
+   in one menu.
+
+   Because of the potential confusion this may cause
+   it is discouraged.
+
+   Example:
+   The following three IDs
+   are defined in a menu, in order:
+   ``test, test2, test``.
+
+   The item with the second "test" ID
+   can then be targeted
+   using the following ID combinations:
+   ``test, test2, test``; ``test, test`` or ``test2, test``.
+
+
+Sub-Menus

```

Comment:
Okay, so we used `subdirectory` once in `basic_concepts.rst` and `submenu` in the file you listd already. Other than that, no other occurences.

I still favor the hyphenated versions, just like with `auto-*`, so I vote for adjusting these instead.